sources/academy/webscraping/scraping_basics_python/02_devtools_locating_elements.md

```diff
@@ -6,12 +6,210 @@ sidebar_position: 2
+
+![Warehouse store with DevTools open](./images/devtools-warehouse.png)
+
+The page displays a grid of product cards, each showing a product's name and picture. Open DevTools and locate the name of the **Sony SACS9 Active Subwoofer**. Highlight it in the **Elements** tab by clicking on it.
+
+![Selecting an element with DevTools](./images/devtools-product-name.png)
+
+Next, let's find all the elements containing details about this subwoofer—its price, number of reviews, image, and more.
+
+In the **Elements** tab, move your cursor up from the `a` element containing the subwoofer's name. On the way, hover over each element until you highlight the entire product card. Alternatively, use the arrow-up key. The `div` element you land on is the **parent element**, and all nested elements are its **child elements**.
+
+![Selecting an element with hover](./images/devtools-hover-product.png)
+
+At this stage, we could use the **Store as global variable** option to send the element to the **Console**. While helpful for manual inspection, this isn't something a program can do.
+
+Scrapers typically rely on [CSS selectors](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_selectors) to locate elements on a page, and these selectors often target elements based on their `class` attributes. The product card we highlighted has markup like this:
+
+```html
+<div class="product-item product-item--vertical 1/3--tablet-and-up 1/4--desk">
+  ...
+</div>
+```
+
+The `class` attribute can hold multiple values separated by whitespace. This particular element has four classes. Let's move to the **Console** and experiment with CSS selectors to locate this element.
+
+## Programmatically locating a product card
+
+Let's jump into the **Console** and write some JavaScript. Don't worry—you don't need to know the language, and yes, this is a helpful step on our journey to creating a scraper in Python.
+
+In browsers, JavaScript represents the current page as the [`Document`](https://developer.mozilla.org/en-US/docs/Web/API/Document) object, accessible via `document`. This object offers many useful methods, including [`querySelector()`](https://developer.mozilla.org/en-US/docs/Web/API/Document/querySelector). This method takes a CSS selector as a string and returns the first HTML element that matches. Try typing this into the **Console**:
+
+```js
+document.querySelector('.product-item');
+```
+
+It will return the HTML element for the first product card in the listing:
+
+![Using querySelector() in DevTools Console](./images/devtools-queryselector.png)
+
+:::note About the missing semicolon
+
+In the screenshot, there is a missing semicolon `;` at the end of the line. In JavaScript, semicolons are optional, so it doesn't make a difference here.
+
+:::
```

Comment:
Since semicolons are optional in Python as well, wouldn't it be easier to just delete the semicolon from the code example - to align it with the screenshot? Rather than writing this long explanation? 😅